### PR TITLE
Add dartdoc for the painting code

### PR DESCRIPTION
**Pull request by abarth**
@Hixie

### CODE REVIEW

File: sky/packages/sky/lib/painting/box_painter.dart [hunk 1]
```diff
@@ -10,22 +10,40 @@ import 'package:sky/base/image_resource.dart';
 import 'package:sky/base/lerp.dart';
 import 'package:sky/painting/shadows.dart';
 
+/// An immutable set of offsets in each of the four cardnal directions
```
Comment (Hixie) [inline sky/packages/sky/lib/painting/box_painter.dart]: cardinal

File: sky/packages/sky/lib/painting/box_painter.dart [hunk 2]
```diff
@@ -10,22 +10,40 @@ import 'package:sky/base/image_resource.dart';
...
+///
+/// Typically used for an offset from each of the four sides of a box. For
+/// example, the padding inside a box can be represented using this class.
 class EdgeDims {
-  // used for e.g. padding
+  // TODO(abarth): Remove this constructor. The `only` constructor is better.
```
Comment (Hixie) [inline sky/packages/sky/lib/painting/box_painter.dart]: or rename it to EdgeDims.TRBL

File: sky/packages/sky/lib/painting/box_painter.dart [hunk 3]
```diff
@@ -139,68 +184,94 @@ class BoxShadow {
...
 class LinearGradient extends Gradient {
   LinearGradient({
-    this.endPoints,
+    this.start,
+    this.end,
     this.colors,
-    this.colorStops,
+    this.stops,
     this.tileMode: sky.TileMode.clamp
   });
```
Comment (Hixie) [inline sky/packages/sky/lib/painting/box_painter.dart]: assert that the lengths are the same

File: sky/packages/sky/lib/painting/box_painter.dart [hunk 4]
```diff
@@ -139,68 +184,94 @@ class BoxShadow {
...
+  /// Note: This list must have the same length as [colors].
+  final List<double> stops;
+
+  /// How this gradient should tile the plane
   final sky.TileMode tileMode;
 
   sky.Shader createShader() {
-    return new sky.Gradient.linear(this.endPoints, this.colors,
-                                   this.colorStops, this.tileMode);
+    return new sky.Gradient.linear([start, end], this.colors,
```
Comment (Hixie) [inline sky/packages/sky/lib/painting/box_painter.dart]: you can remove the other "this"s

File: sky/packages/sky/lib/painting/text_style.dart [hunk 5]
```diff
@@ -4,23 +4,118 @@
...
-enum FontStyle { normal, italic, oblique }
+/// Whether to slant the glyphs in the font
+enum FontStyle {
+  /// Use the upright glyphs
+  normal,
+
+  /// Use glyphs designed for slanting
+  italic,
+
+  /// Use the upright glyphs but slant them during painting
```
Comment (Hixie) [inline sky/packages/sky/lib/painting/text_style.dart]: is that actually what we do?

**Comment by Hixie** [comment]
LGTM

File: sky/packages/sky/lib/painting/text_style.dart (hunk 5 above)
Comment (abarth) [inline sky/packages/sky/lib/painting/text_style.dart]: I believe so...

File: sky/packages/sky/lib/painting/box_painter.dart (hunk 1 above)
Comment (abarth) [inline sky/packages/sky/lib/painting/box_painter.dart]: Done.

File: sky/packages/sky/lib/painting/box_painter.dart (hunk 2 above)
Comment (abarth) [inline sky/packages/sky/lib/painting/box_painter.dart]: Updated comment.

File: sky/packages/sky/lib/painting/box_painter.dart (hunk 3 above)
Comment (abarth) [inline sky/packages/sky/lib/painting/box_painter.dart]: Done.

File: sky/packages/sky/lib/painting/box_painter.dart (hunk 4 above)
Comment (abarth) [inline sky/packages/sky/lib/painting/box_painter.dart]: Done.

File: sky/packages/sky/lib/painting/text_painter.dart [hunk 6]
```diff
@@ -8,7 +8,7 @@ import 'package:sky/painting/text_style.dart';
 
 export 'package:sky/painting/text_style.dart';
 
-// This must be immutable, because we won't notice when it changes
+/// An immutable span of text
```
Comment (abarth) [inline sky/packages/sky/lib/painting/text_painter.dart]: Restored

**Pull request merged**